### DISCUSSION TITLE
Merge Queue: Access to the base_sha of the predictive branch

### DISCUSSION THREAD

**Discussion by bmcmanus-apex** [community/community#41939] [Actions]
### Select Topic Area

Product Feedback

### Body

My company has recently investigated the use of merge queues.  We operate out of a monorepo using Bazel to build.  We only want to build/test targets that have changed in a given PR.  To enable this we hash the Bazel targets for the base_ref and the head_ref and find changed hashes.  In the payload for merge_group events there is a base_sha.  That base_sha is the base_sha for the base_ref of the merge group.  There is no clean way to access the base_sha of the branch that a run is executing off of.  Luckily GitHub uses that sha in the name of the branches it creates. Example:
`gh-readonly-queue/merge-queue-main/pr-9092-d4947bde60fcdd19655b95c100952ac325351124`
.  It would do wonders if the head_ref of the previous branch could be included in the payload as some predictive_base_sha.  RIght now we are stuck with a very brittle implementation that requires us to parse the branch name.  This leaves us in a difficult spot.  If that naming convention ever changed merge queues would no longer be an option for us as we'd no longer have access to the information needed for CI.

**Comment by janeklb** [comment]
I may have just produced a duplicate of this discussion here: https://github.com/orgs/community/discussions/62219

**Comment by joehorsnell** [comment]
Linking to https://github.com/orgs/community/discussions/65239 for visibility, which has more upvotes
